`Improvement`: Redesign Member selection

julian-wls opened the pull request:
### Problem Description

The member selection when creating or adding members to a conversation was using outdated chips to show selected members.

### Changes

This PR is a follow up PR to to #362 and optimizes the view for adding members to conversations by introducing a new FAB, updated recipient chips with profile pictures and some design tweaks.

### Steps for testing

1. Go to the conversation overview.
2. Create a new conversation and check if adding members still works as expected.
3. Go to a channel/group and check if adding new members works as before.

**Note**
I'm going to do a redesign of the conversation settings and the browse channels screen soon to align it to the look of the new member selection, see #365.

FelberMartin commented on feature/metis/manage-conversations/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/manageconversations/ui/conversation/settings/add_members/ConversationAddMembersScreen.kt:
```diff
@@ -105,31 +116,56 @@ internal fun ConversationAddMembersScreen(
+                            .padding(start = 8.dp),
+                        imageVector = Icons.Default.Done,
+                        contentDescription = null
+                    )
```
All this code feels kind of like a duplication of the changes in `CreatePersonalConversationScreen`. If I'm not mistaken, the only difference is the topbar and the error dialog. Do you think we can somehow avoid the code duplication here?

julian-wls replied:
I created an issue for this, see #376

FelberMartin reviewed (changes requested):
Nice improvement, just some code comments :)

FelberMartin commented on feature/metis/manage-conversations/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/manageconversations/ui/conversation/create_personal_conversation/CreatePersonalConversationScreen.kt:
```diff
@@ -92,9 +106,11 @@ internal fun CreatePersonalConversationScreen(
-                .consumeWindowInsets(WindowInsets.systemBars.only(WindowInsetsSides.Top))
-                .padding(horizontal = Spacings.ScreenHorizontalSpacing),
-            viewModel = viewModel
+                .padding(bottom = padding.calculateBottomPadding())
```
Potentially related: There is a weird padding between the keyboard and the RecipientsList (marked in green in the screenshot). It is also bigger when using the 3-dot navigation

![image](https://github.com/user-attachments/assets/3736d63f-0710-4eb2-8f91-be3cd04e7f32)

julian-wls replied:
> Is there a particular reason why we do not extend the MemberSelection content beyond the navigation bars?

I think it would look strange if a card ends with the bottom screen edge without having some space in between. Since we use .imePadding() here there is also some space between the keyboard and the card. 

This is how it looks like when we get rid of the bottom padding, I don't think it looks as good:
![T](https://github.com/user-attachments/assets/a77fdbaa-9527-4c56-abbd-45a7f83199af)

julian-wls commented on feature/metis/manage-conversations/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/manageconversations/ui/conversation/create_personal_conversation/PotentialRecipientsUi.kt:
```diff
@@ -108,26 +117,39 @@ private fun PotentialRecipientsList(
+            LazyColumn(
+                modifier = modifier
+                    .fillMaxWidth(),
+                contentPadding = PaddingValues(bottom = 90.dp),
```
This wouldn't add enough padding to prevent the FAB from being placed over the last element of the list.

FelberMartin replied:
Ahhh, that's the reason for that padding, got it. Just adding a small code comment about that would be awesome :D

eylulnc reviewed (changes requested):
Its looking good! I only have one small point  😅:

<img width="79" alt="Screenshot 2025-02-04 at 00 55 29" src="https://github.com/user-attachments/assets/8fe16dca-78be-4ac6-b007-872b2cb97036" />

I think padding to start and end should be equal

julian-wls commented on feature/metis/manage-conversations/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/manageconversations/ui/conversation/create_personal_conversation/CreatePersonalConversationScreen.kt:
```diff
@@ -92,7 +92,7 @@ internal fun CreatePersonalConversationScreen(
             ) {
                 Row(
-                    modifier = Modifier.padding(horizontal = 8.dp),
+                    modifier = Modifier.padding(16.dp),
```
I actually already changed the padding to be 16.dp on all sites. Could you check again?

eylulnc replied:
Thanks for the highlight 😊

ahbitaqu reviewed (approved):
Works as it should over here

julian-wls commented:
As mentioned after the meeting today I fixed the padding issue by adding the imePadding to the lazyList only and added check mark icons to selected filterChips.

FelberMartin reviewed (approved):
Looking good! As soon as the last open comment is resolved, feel free to add the "ready to merge" label :D
